sprints/dashboard/tasks.py

```diff
@@ -237,17 +238,18 @@ def complete_sprint_task(board_id: int) -> None:
             )
 
             # Ensure that the next sprint exists. If it doesn't exist, create it.
-            future_next_sprint = get_next_sprint(sprints, next_sprint)
-            if not future_next_sprint:
-                create_next_sprint_task(board_id)
-                future_next_sprint = get_next_sprint(sprints, next_sprint)
+            # Get next sprint number for creating role tasks there.
+            if future_next_sprint := get_next_sprint(sprints, next_sprint):
+                future_next_sprint_number = get_sprint_number(future_next_sprint)
+            else:
+                future_next_sprint_number = create_next_sprint_task(board_id)
```

Comment:
That fixes a bug that we encountered once. After some optimizations we started using cached sprints to speed things up, but forgot to refresh it here.